lib/Sema/SemaOverload.cpp

```diff
@@ -1248,7 +1253,7 @@ TryUserDefinedConversion(Sema &S, Expr *From, QualType ToType,
         // Turn this into a "standard" conversion sequence, so that it
         // gets ranked with standard conversion sequences.
         DeclAccessPair Found = ICS.UserDefined.FoundConversionFunction;
-        ICS.setStandard();
+        ICS.setStandard(ImplicitConversionSequence::MemsetToZero);
```

Comment:
I'm not sure any of these are needed if you simply write `= 0` to initialise your new field.